### PR TITLE
RSP-1793 [Accessibility][Docs] Add FocusRing to SideNav items and Links

### PR DESCRIPTION
**Pull request by majornista**
Relates to https://jira.corp.adobe.com/browse/RSP-1793 and #527

Use `FocusRing` or `Link` rather than define style overrides as done here: https://github.com/adobe-private/react-spectrum-v3/pull/527/commits/5af13daeebbced78b8db8289e1f863622c48a9b3

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [RSP-1793](https://jira.corp.adobe.com/browse/RSP-1793) and #527.
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

`yarn start:docs`

Navigate through site using the tab key, all links should now display the FocusRing style.

## 🧢 Your Project:

Accessibility

### CODE REVIEW

**Comment by snowystinger** [comment]
When I tab through these links, they sometimes go in the wrong order
Screen Shot 2020-06-04 at 2 35 40 PM

**Comment by snowystinger** [comment]
o, not wrong order, just sometimes the double underline doesn't render

**Comment by snowystinger** [comment]
side nav seems to be working great :)

File: packages/dev/docs/src/addFocusRing.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import {FocusRing} from '@react-aria/focus';
+import {Link} from '@react-spectrum/link';
+import linkStyles from '@adobe/spectrum-css-temp/components/link/vars.css';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import sideNavStyles from '@adobe/spectrum-css-temp/components/sidenav/vars.css';
+
+export function addFocusRing() {
+  let links = document.getElementsByClassName(sideNavStyles['spectrum-SideNav-itemLink']);
```
Comment (ross-pfahler) [inline packages/dev/docs/src/addFocusRing.js:9]: for the sidenav links does it make sense to add this code here? https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/dev/docs/src/Layout.js#L234-L240

or is this something that only should happen clientside?

Comment (majornista) [reply packages/dev/docs/src/addFocusRing.js:9]: I think this has to render client side for the React-Spectrum Link to be available. @snowystinger?

Comment (snowystinger) [reply packages/dev/docs/src/addFocusRing.js:9]: yeah, these things won't do anything if they are ssr'ed, no interactions, just dom. so to get the focus ring to work, it needs to have actual runnable code on the client, hence this

Comment (ross-pfahler) [reply packages/dev/docs/src/addFocusRing.js:9]: got it. bummer.

**Comment by majornista** [comment]
> o, not wrong order, just sometimes the double underline doesn't render

I've noticed that too, at least in Chrome, and think it's a CSS rendering problem. The FocusRing class is being applied correctly.

**Comment by dannify** [comment]
Covered by #658

**Pull request closed without merging**